crates/shared/src/price_estimation/instrumented.rs

```diff
@@ -42,7 +30,20 @@ impl PriceEstimating for InstrumentedPriceEstimator {
         &'a self,
         queries: &'a [Query],
     ) -> futures::stream::BoxStream<'_, (usize, super::PriceEstimateResult)> {
-        old_estimator_to_stream(self.estimates_(queries))
+        let start = Instant::now();
+        let measure_time = async move {
+            self.metrics
+                .price_estimation_timed(&self.name, start.elapsed());
+        };
+        self.inner
+            .estimates(queries)
+            .map(move |result| {
+                let success = !matches!(&result.1, Err(PriceEstimationError::Other(_)));
+                self.metrics.price_estimated(&self.name, success);
+                result
+            })
```

Comment:
nit: Seems like this would be a good use for [`inspect`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.inspect)

Reply:
Changed